frontend/src/components/set-org/SetOrg.jsx

```diff
@@ -23,23 +25,29 @@ function SetOrg() {
   const handleException = useExceptionHandler();
 
   useEffect(() => {
-    if (state === null || signedInOrgId) {
-      navigate("/");
-    }
-  }, []);
+    const fetchData = async () => {
```

Comment:
@muhammad-ali-e Can we declare this function outside the `useEffect` hook?

Reason: The function will be recreated each time the `useEffect` hook runs. In our case, each time the `state` or `navigate` value changes, this `useEffect` hook will get triggered, and the `fetchData` function will be recreated.

Reply:
@tahierhussain I tried But it have to be inside useEffect
https://react.dev/reference/react/useEffect#fetching-data-with-effects:~:text=a%20cleanup%20function%3A-,App.js,-Reset

Reply:
@muhammad-ali-e Let me look into this and we can address this improvement in a separate PR. I'm resolving this conversation and approving this PR so that it's not blocked.

cc: @nehabagdia